rs/consensus/src/consensus/bounds.rs

```diff
@@ -48,25 +48,26 @@ fn get_maximum_validated_artifacts(node_count: usize, dkg_interval: usize) -> Ar
     let e = MINIMUM_CHAIN_LENGTH as usize;
     /*
      * To derive our bounds, we consider a worst-case scenario in which we have
-     * notarizations (or finalizations) for d rounds into the DKG interval i, without
-     * having a CUP for it yet. In this case, we keep all artifacts from the (i-1)-th
-     * interval, including a minimum chain length of `e` from the (i-2)-th interval.
+     * notarizations (or finalizations) for d rounds above the i-th summary
+     * height, without having a CUP for it yet. In this case, we keep all
+     * artifacts from the (i-1)-th interval, including a minimum chain length
+     * of `e` from the (i-2)-th interval.
      *
-     *       [B] Ordinary blocks                            notarized/finalized
+     *       [B] Ordinary blocks                             notarized/finalized
      *       [S] Summary blocks                                   height
      *                                                              |
      *  ..-[B]-[B]-...-[B]-[S]-[B]-[B]-...-[B]-[B]-[S]-[B]-...-[B]-[B] <- chain tip
-     *      |               |                       |               |
-     *      +--- minimum  --+---- DKG interval -----+-- d rounds ---+
-     *      |  chain length           (k)                           |
-     *      |      (e)                                              |
-     *      +------------------- l = e + k + d ---------------------+
+     *      |           |   |                   |   |               |
+     *      +- minimum -+   +-- DKG interval ---+   +- d+1 heights -+
+     *      |chain length           (k)                             |
+     *      |    (e)                                                |
+     *      +--------------- l = e + k + (d + 1) -------------------+
      *
      * In the above scenario, assuming `d` is equal to the maximum notarization/CUP
      * gap, nodes will refuse to notarize any further blocks. So `l` is the maximum
-     * height span we need to consider for placing a bound on the artifact counts.
+     * number of rounds we need to consider for placing a bound on the artifact counts.
      */
-    let l = k + e + d;
+    let l = k + e + (d + 1);
```

Comment:
Is the problem more generally that the inclusive range `[x-l, x]` consists of l+1 heights? Or why is it exclusive to `d`?

Are we now allowing one artifact too many, where ever we used `(l+1)` below?

Reply:
Yes, the problem is the inclusive range. I updated the illustration, and marked the heights we care about with `h0, h1, ...`.